src/common/snippets/src/pass/lowered/loop_init.cpp

```diff
@@ -0,0 +1,234 @@
+                  std::vector<LoweredExprPtr>& loop_in_exprs, std::vector<LoweredExprPtr>& loop_out_exprs,
+                  OutputVector& loop_in_outputs, OutputVector& loop_out_outputs) {
+    loop_in_exprs.clear();
+    loop_out_exprs.clear();
+    loop_in_outputs.clear();
+    loop_out_outputs.clear();
+
+    for (const auto& loop_entry_point : loop_entries) {
+        const auto expr = loop_entry_point.first;
+        const auto node = expr->get_node();
+        if (is_type<op::Load>(node) || is_type<op::BroadcastLoad>(node)) {
+            // Todo: Sometimes several Load in one Loop read data from the same Node.
+            if (std::none_of(loop_in_outputs.begin(), loop_in_outputs.end(),
+                             [node](const ov::Output<ov::Node>& out) { return node->get_input_node_shared_ptr(0) == out.get_node_shared_ptr(); })) {
+                loop_in_outputs.push_back(node->input_value(0));
+                loop_in_exprs.push_back(expr);
+            }
+        }
+    }
+
+    for (const auto& loop_exit_point : loop_exits) {
+        const auto expr = loop_exit_point.first;
+        const auto node = expr->get_node();
+        if (is_type<op::Store>(node)) {
+            const auto &dest = node->output(0);
+            loop_out_outputs.push_back(dest);
+            loop_out_exprs.push_back(expr);
+        }
+    }
+}
+
+int64_t get_dim_stride(const size_t dim, const std::vector<size_t>& layout, const std::vector<size_t>& shape) {
+    int64_t stride = 1;
+    for (int i = static_cast<int>(layout.size()) - 1; i >= 0; i--) {
+        if (layout[i] == dim)
+            break;
+        stride *= static_cast<int64_t>(shape[layout[i]]);
+    }
+    return stride;
+}
+}  // namespace
+
+LoopInit::LoopInit() : LinearIRTransformation() {}
+
+std::vector<int64_t> LoopInit::init_ptr_increments(LoweredExprIR& linear_ir,
+                                                   const std::vector<LoweredExprPtr>& loop_in_exprs,
+                                                   const std::vector<LoweredExprPtr>& loop_out_exprs,
+                                                   size_t dim_idx) const {
+    std::vector<int64_t> ptr_increments;
+    // Note: All loop inputs must have the same layout by definition.
+    // If this doesn't hold, then we're trying to inject loops in the wrong place.
+    const std::vector<size_t> loop_layout{
+            !loop_in_exprs.empty() ? loop_in_exprs.front()->get_inputs()[0]->get_layout() :
+            !loop_out_exprs.empty() ? loop_out_exprs.front()->get_outputs()[0]->get_layout() :
+            std::vector<size_t>{}};
+    // Note: Need to find max relevant dim first to account for broadcasting, collect relevant_dims as well
+    size_t max_relevant_dim_size = 0;
+    for (const auto& expr : loop_in_exprs) {
+        const auto& out_tds = expr->get_outputs();
+        const auto& dst_layout = out_tds[0]->get_layout();
+        const auto& dst_tensor = out_tds[0]->get_tensor();
```

Comment:
This code looks familiar)
But seriously, now when we have `LoweredExprPort` it seems that we need to check relevant td, not just `out_tds[0]`. Otherwise sooner or later we'll get an example when this heuristics doesn't work)